packages/artifact/src/internal/shared/artifact-twirp-client.ts

```diff
@@ -76,6 +76,9 @@ class ArtifactHttpClient implements Rpc {
     let errorMessage = ''
     while (attempt < this.maxAttempts) {
       let isRetryable = false
+      let isDuplicateArtifactName = false
+      let isTooManyArtifacts = false
```

Comment:
I wonder if these variables are needed - or if in the error block you could do something like:

```
if (this.isAlreadyExistsStatusCode(statusCode)) {
  throw new Error.....
}
```

I also thought the service today has an interesting approach to making a single function that maps error codes, which might consolidate some code: https://github.com/actions/toolkit/blob/e8f21993484c802107e8c4c3839e86b0b6e8954b/packages/artifact/src/internal/upload-http-client.ts#L93-L94